packages/react/src/Grid/Grid.tsx

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license EUPL-1.2+
+ * Copyright (c) 2023 Gemeente Amsterdam
+ */
+
+import clsx from 'clsx'
+import {
+  ForwardedRef,
+  forwardRef,
+  ForwardRefExoticComponent,
+  HTMLAttributes,
+  PropsWithChildren,
+  RefAttributes,
+} from 'react'
+import { GridCell } from './GridCell'
+
+export type GridColumnNumber = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12
+
+export type GridColumnNumbers = {
+  narrow: GridColumnNumber
+  medium: GridColumnNumber
+  wide: GridColumnNumber
+}
+
+interface GridComponent
+  extends ForwardRefExoticComponent<PropsWithChildren<HTMLAttributes<HTMLDivElement>> & RefAttributes<HTMLDivElement>> {
+  Cell: typeof GridCell
+}
+
+export const Grid = forwardRef(
+  (
+    { children, className, ...restProps }: PropsWithChildren<HTMLAttributes<HTMLDivElement>>,
+    ref: ForwardedRef<HTMLDivElement>,
+  ) => (
+    <div {...restProps} ref={ref} className={clsx('amsterdam-grid', className)}>
+      {children}
+    </div>
+  ),
+) as GridComponent
+
+Grid.Cell = GridCell
+Grid.displayName = 'Grid'
+GridCell.displayName = 'Grid.Cell'
```

Comment:
Funny that you would have to declare this twice, also in GridCell, I'm not sure what the correct method should be it's hard to find "official" documentation on compound components. Perhaps the direction you're going is too far and it should look like:
```
export default Grid
export { GridCell }
```
Feel free to resolve this brainfart.